src/NHibernate.Test/TransactionTest/TransactionFixtureBase.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections;
+
+namespace NHibernate.Test.TransactionTest
+{
+	public abstract class TransactionFixtureBase : TestCase
+	{
+		protected override IList Mappings => new[] { "TransactionTest.Person.hbm.xml" };
+
+		protected override string MappingsAssembly => "NHibernate.Test";
+
+		protected override void OnTearDown()
+		{
+			using (var s = OpenSession())
+			{
+				s.CreateQuery("delete from System.Object").ExecuteUpdate();
+			}
```

Comment:
I wonder whether such `Teardown` should be directly added in `TestCase`. It would catch all cases not requiring ordering for deletes, and could still be overridden by test needing a specific delete ordering.